src/libutil/error.hh

```diff
@@ -50,6 +50,14 @@ struct LinesOfCode {
     std::optional<std::string> nextLineOfCode;
 };
 
+/* NOTE: position.hh recursively depends on source-path.hh -> source-accessor.hh
+   -> hash.hh -> config.hh -> experimental-features.hh -> error.hh -> Pos.
+   There are other such cycles.
+   Thus, Pos has to be an incomplete type in this header. But since ErrorInfo/Trace
+   have to refer to Pos, they have to use pointer indirection via std::shared_ptr
+   to break the recursive header dependency.
+   FIXME: Untangle this mess. Should there be AbstractPos as there used to be before
+   4feb7d9f71? */
```

Comment:
I am guessing `experimental-features.hh` is the easy thing to make more abstract

I was planning on doing that anyways for a new settings system.

Reply:
Forgot to write down more thoughts.

While it would be good to slim down `nix-util` and have more things in libraries with well-defined scopes, I think it's ok to consider `Pos` to be part of a text handling component; not the evaluator. This frees it up to be used in `nix.conf` logic and other file types.